sklearn/metrics/_decision_threshold.py

```diff
@@ -0,0 +1,117 @@
+"""Metric per threshold curve to assess binary classification performance.
+
+Compute metric per threshold, over a range of threshold values to aid visualization
+of threshold-dependent metric behavior.
+
+Utilizes `_CurveScorer` methods to do all the computation.
+"""
+
+# Authors: The scikit-learn developers
+# SPDX-License-Identifier: BSD-3-Clause
+
+from numbers import Integral
+
+from ..utils._param_validation import Interval, validate_params
+
+
+@validate_params(
+    {
+        "score_func": [callable],
+        "y_true": ["array-like"],
+        "y_score": ["array-like"],
+        "thresholds": [
+            Interval(Integral, 2, None, closed="left"),
+            "array-like",
+        ],
+        "greater_is_better": ["boolean"],
+        "labels": ["array-like", None],
+    },
+    prefer_skip_nested_validation=True,
+)
+def decision_threshold_curve(
+    score_func,
+    y_true,
+    y_score,
+    # Should below 2 have a default value?
+    thresholds=20,
+    greater_is_better=True,
+    labels=None,
+    **kwargs,
+):
+    """Compute threshold-dependent metric of interest per threshold.
+
+    Note: this implementation is restricted to the binary classification task.
+
+    Read more in the :ref:`User Guide <threshold_tunning>`.
+
+    .. versionadded:: 1.8
+
+    Parameters
+    ----------
+    score_func : callable
+        The score function to use. It will be called as
+        `score_func(y_true, y_pred, **kwargs)`.
+
+    y_true : array-like of shape (n_samples,)
+        Ground truth (correct) target labels.
+
+    y_score : array-like of shape (n_samples,)
+        Continuous response scores.
+
+    thresholds : int or array-like, default=20
+        Specifies number of decision thresholds to compute score for. If an integer,
+        it will be used to generate `thresholds` thresholds uniformly distributed
+        between the minimum and maximum of `y_score`. If an array-like, it will be
+        used as the thresholds.
+
+    greater_is_better : bool, default=True
+        Whether `score_func` is a score function (default), meaning high is
+        good, or a loss function, meaning low is good. In the latter case, the
+        the output of `score_func` will be sign-flipped.
+
+    labels : array-like, default=None
```

Comment:
`_CurveScorer` uses the term "classes" but "labels" is consistent with what is used for other classification metrics, so chose this.